apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/atoms.ts

```diff
@@ -245,6 +281,28 @@ export const toggleBtcOrSatValueActionAtom = atom(
   }
 )
 
+export const toggleFiatCurrencyActionAtom = atom(
+  null,
+  (get, set, currency: CurrencyCode) => {
+    const btcInputValue = get(btcInputValueAtom)
+    const btcPriceData = get(btcPriceDataAtom)[currency]
+
+    set(selectedCurrencyCodeAtom, currency)
+
+    if (!btcPriceData || btcPriceData.state === 'error') {
+      void set(refreshCurrentBtcPriceActionAtom)()
```

Comment:
Why is this called only on error? Why not call `set(refreshCurrentBtcPriceActionAtom)()` every time and let the caching take care of not sending too many requests?

Even the logics in these atoms are the same.

Reply:
I get that now. We don't expect to spend enough time on this screen to make the price significally outdated. 

But I think calling just `void set(refreshCurrentBtcPriceActionAtom)()` (removing the condition and the else if branch) won't change the functionality, right?

Reply:
Yes it will not change the functionality, this fetch is needed as the user can change currency to some, that has not been fetched before. So calling just refresh without conditions is OK I think